pkg/binding/to_event.go

```diff
@@ -87,7 +87,7 @@ func (b *messageToEventBuilder) SetData(data io.Reader) error {
 		return err
 	}
 	if w != 0 {
-		return b.event.SetData(buf.Bytes())
+		b.event.DataEncoded = buf.Bytes() // TODO: this might be wrong for base64.
```

Comment:
I think you can access the `ContentType` directly and use `SetData`, like:

```
ct := b.event.GetDataContentType()
// Maybe check if ct == "" and apply a default value
b.event.SetData(ct, buf.Bytes())
```

Reply:
that will not do what you want because  `buf.Bytes()` is a `[]byte` and SetData will think you want base64 encoded data out

Reply:
Ok i'm fine leaving as is